src/util/cluster.js

```diff
@@ -13,6 +14,10 @@ function finalFunction() {
   RedisDB.disconnect();
   logger.info('Redis client disconnected');
 
+  terminatePiscina().catch((error) => {
```

Comment:
Why not create a `const` with the `process.env.USE_PISCINA` on top and then check that here to see if we really need to call `initializePiscina` and `terminatePiscina`? It seems better than accessing env vars from everywhere and making methods idempotent in case they are called when they are not supposed to.

Reply:
Made this change. Marking resolved